### PR TITLE
Karo dev

### PR DESCRIPTION
**Pull request by krelo1320**
issue-1: Translated input view to ts

### CODE REVIEW

**Review by oneacik — COMMENTED** [review]
Please gitignore all files that come from your IDE.

Also pls make this pull request contain only one change.
It will be more readable and simple for others.

Please also set title to contain:
issue-<number of task in trello>: <short description>

**Comment by krelo1320** [comment]
I did what you said plus I renamed chart file so it wont collide with Jarek's

**Comment by oneacik** [comment]
Please use correct naming for pull request and commits if possible.
You can take a look on this or:
https://github.com/Krzelaki/js-cashburn/pull/5/commits

You can take issue-* numbers from task URLs, for example:
https://trello.com/c/KgOFv7dV/7-make-memoryservice-persist-data

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import NameForm from "./components/NameForm";
```
Comment (oneacik) [inline src/App.js:2]: Please revert it to NameForm

File: src/components/NameForm1.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+type PROPS = {callback: (arg:number) => void};
```
Comment (oneacik) [inline src/components/NameForm1.tsx]: Lets use camelCase here too:
Props, State.

File: src/components/NameForm1.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
```
Comment (oneacik) [inline src/components/NameForm1.tsx]: Please format the code :P
Everywhere

File: src/components/NameForm1.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+class NameForm extends React.Component<PROPS, STATE> {
+  handleChange: (event: any) => void;
+  handleSubmit: () => void;
+  state: { value: number };
+
+  constructor(props: PROPS) {
+    super(props);
+    this.state = {value: 0};
+
```
Comment (oneacik) [inline src/components/NameForm1.tsx]: nit: use one enter, not too.

File: src/components/NameForm1.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <div>
+          <label>
+            Podaj kwotę:
+            <input
+                className="input"
+                type="number"
+                value={this.state.value}
+                onChange={this.handleChange}/>
+            </label>
```
Comment (oneacik) [inline src/components/NameForm1.tsx]: make the label contain only the label.

File: src/components/NameForm1.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+          <button className = "button" onClick = {() => this.handleSubmit()}>Zatwierdź</button>
```
Comment (oneacik) [inline src/components/NameForm1.tsx]: u can use this.handleSubmit instead of ()=>this.handleSubmit()

**Comment by krelo1320** [comment]
Done! I made it look stylish as you wish

File: src/components/Chart_M.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
```
Comment (oneacik) [inline src/components/Chart_M.js]: Do we need Magda's changes here?
Please keep it granular (one thing as a time), so pull request is easy to understand.

File: src/components/NameForm.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+type Props = { callback: (arg: number) => void };
+type State = { value: number };
+
+class NameForm extends React.Component<Props, State> {
+    handleChange: (event: any) => void;
```
Comment (oneacik) [inline src/components/NameForm.tsx]: Please find the type of event.

File: src/core/Memory.test.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { getValues, saveValues } from "./Memory";
```
Comment (oneacik) [inline src/core/Memory.test.tsx]: duplicated test.
You don't  need to use JSX here.

File: src/core/Memory.tsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+let values_in_browser = []
```
Comment (oneacik) [inline src/core/Memory.tsx]: another duplication

File: src/index.html [hunk 11]
```diff
@@ -5,6 +5,7 @@
       <title>React App</title>
    </head>
    <body>
+<script src = "components/bienvenidos.js"></script>
```
Comment (oneacik) [inline src/index.html]: pls remove

File: src/components/NameForm.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
+import React, {ChangeEventHandler, SyntheticEvent} from "react";
+import ReactDOM from "react-dom";
+
+type Props = { callback: (arg: number) => void };
+type State = { value: number };
+
+class NameForm extends React.Component<Props, State> {
+    handleChange: ChangeEventHandler<HTMLInputElement>;
+    //handleChange: {event: SyntheticEvent<Element, Event>};
+    handleSubmit: () => void;
```
Comment (oneacik) [inline src/components/NameForm.tsx]: make this handleSubmit typed as *EventHandler.
Probably SubmitEventHandler or something like that.

File: src/components/NameForm.tsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    state: { value: number };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {value: 0};
+
+        this.handleChange = (event) => this.setState({value: parseInt(event.target.value)});
+        this.handleSubmit = () => {
+            props.callback(this.state.value)
```
Comment (oneacik) [inline src/components/NameForm.tsx]: You don't need to use '{}' if you have only one expression.
e.g () => props.callback(this.state.value)

File: src/components/NameForm.tsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    handleChange: ChangeEventHandler<HTMLInputElement>;
+    //handleChange: {event: SyntheticEvent<Element, Event>};
+    handleSubmit: () => void;
+    state: { value: number };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {value: 0};
+
+        this.handleChange = (event) => this.setState({value: parseInt(event.target.value)});
```
Comment (oneacik) [inline src/components/NameForm.tsx:16]: same here

**Pull request closed without merging**